priv/static/assets/js/arizona-worker.js

```diff
@@ -111,32 +111,39 @@ function handleEvent(data) {
   }
 }
 
-function applyPatch([target, changes]) {
-  const tree =
-    target === "root" ? state.tree : getTargetTree(target, state.tree);
+function applyPatch([[, ...target], changes]) {
+  const tree = getTargetTree(target, state.tree[1]);
   changes.forEach((c) => {
     applyChanges(c, tree);
   });
-  const html = tree.flat(Infinity).join("");
-  return { target, html };
+  const html = zip(tree[0], tree[1]);
+  return { target: [0, ...target], html };
+}
```

Comment:
A brief note here...
Now, block IDs always start with zero, and zero is the "root"/main block.
The server needs this information to handle sockets and renders, but the client "patch" doesn't.
What happens here is: I'm suppressing the zero in the function args and prefixing it to the return of the function in the 'target' prop to get the correct element in the DOM.
Currently, I don't have a solution that works on both sides without this "workaround" on the client side.
Anyway, it works :smile: